keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/PublishedArtifact.kt

```diff
@@ -91,6 +91,15 @@ data class PublishedArtifact(
         }
       }
 
+  val branch: String?
+    get() = gitMetadata?.branch ?: metadata["branch"] as? String
+
+  val commitHash: String?
+    get() = gitMetadata?.commit ?: metadata["commitId"] as? String
+
+  val buildNumber: String?
+    get() = buildMetadata?.number ?: metadata["buildNumber"] as? String
+
```

Comment:
I found that Igor already provided all this metadata when we first get information about the artifacts (i.e. before we call again to retrieve additional git and build metadata). These accessors expose the corresponding bits of metadata delegating to the generic `metadata` from Igor first, and then fallback to the `gitMetadata` and `buildMetadata`.

Reply:
Correction: the only thing missing was the branch, which I've added in Igor.